File: java/868.java
class Solution {
    public int binaryGap(int N) {
        int left = -1;
        int i = 0,ans = 0;
        while(N > 0){
            int k = N % 2;
            N /= 2;
            if(k == 1){
                if(left == -1){
                    left = i;
                }else{
                    ans = Math.max(i-left,ans);
                    left = i;
                }
            }
            i ++;
        }
        return ans;
    }
}

File: go/868.go
func binaryGap(N int) int {
	left := -1
	ans := 0
	i := 0
	for N > 0 {
		k := N % 2
		N /= 2
		if k == 1 {
			if left == -1 {
				left = i
			} else {
				if i-left > ans {
					ans = i - left
				}
				left = i
			}
		}
		i++
	}
	return ans
}